### ISSUE TITLE
Suggestion: Allow filtering out of specific races, etc.

### ISSUE THREAD

**Issue by Shuggaloaf** [adougherty/random-everything-generator#6]
Tired of me yet? Also, how many of my suggestions have to get implemented before I get a free Patreon version? ;-p I kid, I kid. 

<img src="https://user-images.githubusercontent.com/79543184/146301197-b447c27f-6556-4aad-b71a-5df197fef0ba.png" width="30%" />

I noticed there are a lot of race options when I was testing generating the merchant above. This is great, but a lot of us are going to limit random NPCs, say, working in the town tavern or at the guidhall (etc.), to only specific races.  I know you can click the refresh button, but I realized this could use a filter after clicking it 5 times and still not getting a race I would use in my game. Like our werewolf merchant for example. Could lead to some cool concepts, I do not deny. But in 99% of cases I'm just looking for your standard PHB races for shops and things of that nature.

I'm pointing out race in this example, but I'm sure once I dig further into other generator options that there will likely be other fields that could use a filter like this as well. (Be happy to post back with more as I find them if that's helpful).

Not sure how hard that is. If races/whatever-else-is-filterable is just 1 table each, that probably wouldn't be difficult. But if you have something like Merchant_Race; Barkeep_Race; Etc. then I could see this being much more difficult to implement.

**Comment by adougherty** [comment]
I see what you mean with this and I agree. I'm thinking implementing weighted tables would solve this issue. Right now a "Werewolf" is given the same probability of being rolled as a Dwarf. Really, it should be a Dwarf has a 10% chance of being rolled, while a Werewolf has a 1% chance of being rolled.

**Comment by Shuggaloaf** [comment]
>I'm thinking implementing weighted tables would solve this issue. 

My first instinct was that yes, it would, but I'm not entirely sure about that. Reason being is there are a LOT of non-PHB races listed. I bet you will find that you have dozens of these races and that setting them at 1% would not leave much room for other races. Or, maybe I should not allow a high enough probability. So if you went this route, you'd probably have to make it less than 1%. 

Personally I still think allowing the user to remove races they do not want would be the _optimal_ solution. But I understand if this is difficult that it probably won't happen. The other potential issue I see is that there may be other fields aside form race that could also need filtering. 

Another possibility, and just spitballing here, could be to have a "standard" and "expanded" version. But I would see that as probably being a lot more work for you to maintain. 

So, for now, perhaps the best solution is to make those races a <1% probability but also keep this suggestion open for a while as you continue to develop. This would give me and others a chance to add other types of things that could use a filter or probability adjustment. It would allow you to not have to commit to anything major while focusing on getting everything together.

**Comment by adougherty** [comment]
If I understand correctly, you'll looking to set on a global setting (for all of your generated tables), a filtered list of possible race results. I can see where this would be quite useful, especially considering different gaming systems, and different homebrew worlds. It would probably also apply to other system specific attributes such as class or background. I'll add all that to my todo list

**Comment by Shuggaloaf** [comment]
Yes! This is exactly what I meant. When I responded back I thought you understood that but were just proposing probabilities as a solution. Sorry I wasn't clear!

I'm glad you can see how it would be useful. With dozens of different systems and limitless homebrew worlds for those systems, allowing global filters for some things would be extremely useful. And yes, class and background would be 2 more of those "other items" I mentioned. I'm sure as you work on tweaking it and as I play with it more that we can find other good examples of filter items.

**Comment by Shuggaloaf** [comment]
No longer following. Dev may close this if they wish.
